includes/abstracts/abstract-wc-product.php

```diff
@@ -1420,6 +1420,12 @@ public function is_visible() {
 			$visible = false;
 		}
 
+		if ( $this->get_parent_id() ) {
+			if ( ( $parent_product = wc_get_product( $this->get_parent_id() ) ) && 'publish' !== $parent_product->get_status() ) {
```

Comment:
This still needs to come before the `if`: `$parent_product = wc_get_product( $this->get_parent_id()`